### PR TITLE
Project week 16: Labyrinth (Ida Sylvén & Sandra Wallén)

### PR DESCRIPTION
**Pull request by sandra-wallen**
**Link to our deployed project:**
https://trusting-lewin-948ec6.netlify.app/

**Our reflections on how this week's project turned out:**
We are very happy with our project! Two cat persons putting their clever heads together – what could possibly go wrong?! Jokes aside - we have had a really pleasant week working together! 

We have accomplished all Blue Level requirements, as well as some Red and Black Level stretch goals. 

For more details, please see readme file.

### CODE REVIEW

File: code/src/components/StartScreen.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      <NeonCat />
+      <div className="nes-field">
+        <label htmlFor="name_field">Enter your name to start the game:</label>
+        <input 
+          type="text" 
+          id="name_field" 
+          className="nes-input" 
+          value={name} 
+          onChange={onInputChange} 
+        />
```
Comment (IsabellaM5) [inline code/src/components/StartScreen.js:34]: Nice feature with adding a unique id to the username! ✨

One thing we noticed though is that it's possible to enter the game without typing in a username. One suggestion is to make it required (or another kind of validation), and an appropriate message to the user on what to do, preventing the user from proceeding on. 🙂

File: code/src/index.css [hunk 2]
```diff
@@ -5,9 +5,170 @@ body {
...
+  width: 45%;
+}
+
+.option-container p {
+  font-size: 12px;
+}
+
+.options-wrapper {
+  display: flex;
+  flex-direction: row;
```
Comment (IsabellaM5) [inline code/src/index.css:81]: For readability purposes, one suggestion would be to set flex-direction to column when displaying the direction alternatives in mobile view. It looks a bit cramped 😀

File: code/src/reducers/directions.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    step: localStorage.getItem('directions') 
+      ? JSON.parse(localStorage.getItem('directions'))
+      : null, 
+    history: null, 
+    loading: false
+  },
+  reducers: {
+    setPlayer: (store, action) => {
+      const uniqid = require('uniqid')
+      store.name = action.payload
```
Comment (IsabellaM5) [inline code/src/reducers/directions.js:19]: This was a nice and clever idea to implement a functionality like this! It's very user friendly! 👍

**Review by IsabellaM5 — APPROVED** [review]
Great job with this project! You did all blue requirements and also did some stretch goals! We loved the retro theme and the Nyan cat 😍 

As you suggested, it would be nice with a back button allowing the user to move around freely! Your other ideas also sound great! ✨

APPROVED! Check our comments/suggestions below!

/Anna & Isabella 🐱🐱

File: code/src/reducers/directions.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+import { API_START, API_MOVE } from '../reusables/urls'
+
+export const directions = createSlice({
+  name: "directions",
+  initialState: {
+    name: '',
+    player: '',
+    step: localStorage.getItem('directions') 
+      ? JSON.parse(localStorage.getItem('directions'))
```
Comment (KarolCislowski) [inline code/src/reducers/directions.js:11]: nice that you saved game status, but You need to remember to add an easy way to remove it. For example, after finishing the game or on click event in restart button. Your users are not IT people and they will not know how to clean local storage from console level.

**Review by KarolCislowski — APPROVED** [review]
Great app 🥳

**Pull request closed without merging**